src/Controls/src/Core/Platform/Windows/Extensions/FormattedStringExtensions.cs

```diff
@@ -128,5 +129,54 @@ public static void UpdateInlines(this TextBlock textBlock, Label label)
 
 			return Tuple.Create(run, span.TextColor, span.BackgroundColor);
 		}
+
+		internal static void RecalculateSpanPositions(this TextBlock control, Label element, IList<double> inlineHeights)
+		{
+			if (element?.FormattedText?.Spans == null	
+				|| element.FormattedText.Spans.Count == 0)
+				return;
+			
+			var labelWidth = control.ActualWidth;
+
+			if (labelWidth <= 0 || control.Height <= 0)
+				return;
+
+			for (int i = 0; i < element.FormattedText.Spans.Count; i++)
+			{
+				var span = element.FormattedText.Spans[i];
+
+				var inline = control.Inlines.ElementAt(i);
+				var startRect = inline.ContentStart.GetCharacterRect(LogicalDirection.Forward);
```

Comment:
Always returning an empty Rect. Reviewing why is happening and possible workarounds.